### PR TITLE
feat(mutelist): add mute_finding method

### PR DESCRIPTION
**Pull request by pedrooot**
### Description
This pull request introduces a new method `mute_finding` to the `Mutelist` class and adds corresponding unit tests for various cloud providers. The main changes include the addition of the method itself, which checks if a finding should be muted, and the implementation of tests to ensure this functionality works correctly across AWS, Azure, GCP, and Kubernetes.

### New Method Addition:
* Added `mute_finding` method to `Mutelist` class in `prowler/lib/mutelist/mutelist.py` to check and update the status of a finding based on muting conditions. (`[prowler/lib/mutelist/mutelist.pyR241-R271](diffhunk://#diff-09b4cefd150351713ac7e94d472eb048baf9cc26b2d9672b14159e1b6d8cd522R241-R271)`)

### Unit Tests:
* Added `test_mute_finding` for AWS in `tests/providers/aws/lib/mutelist/aws_mutelist_test.py` to verify the `mute_finding` method. (`[tests/providers/aws/lib/mutelist/aws_mutelist_test.pyR1847-R1880](diffhunk://#diff-a2ee83f49b2b18e5c8290d2289239c2abc8c0d0c2b7d2c46833517daf4f07f17R1847-R1880)`)
* Added `test_mute_finding` for Azure in `tests/providers/azure/lib/mutelist/azure_mutelist_test.py` to verify the `mute_finding` method. (`[tests/providers/azure/lib/mutelist/azure_mutelist_test.pyR69-R102](diffhunk://#diff-63327ab7d6943494eca0591588ec8396a503556da00a9ebc8e64f672ab97cba7R69-R102)`)
* Added `test_mute_finding` for GCP in `tests/providers/gcp/lib/mutelist/gcp_mutelist_test.py` to verify the `mute_finding` method. (`[tests/providers/gcp/lib/mutelist/gcp_mutelist_test.pyR67-R100](diffhunk://#diff-7db0a0db7d0cb555eb4e03dac8415eb978bea40c1bc3186bac714326cb46e292R67-R100)`)
* Added `test_mute_finding` for Kubernetes in `tests/providers/kubernetes/lib/mutelist/kubernetes_mutelist_test.py` to verify the `mute_finding` method. (`[tests/providers/kubernetes/lib/mutelist/kubernetes_mutelist_test.pyR131-R164](diffhunk://#diff-c3e8768eecca04e9e042d229c49331423bba0e9fe54c624c2f04f03049c3886fR131-R164)`)

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/mutelist/mutelist.py [hunk 1]
```diff
@@ -237,6 +238,37 @@ def is_muted_in_check(
...
+        try:
+            is_finding_muted = self.is_muted(
+                finding.account_uid,
+                finding.metadata.CheckID,
+                finding.region,
+                finding.resource_id,
+                unroll_dict(unroll_tags(finding.resource_tags)),
+            )
+
+            if is_finding_muted:
```
Comment (jfagoagas) [inline prowler/lib/mutelist/mutelist.py]:
```suggestion
            if self.is_muted(
                finding.account_uid,
                finding.metadata.CheckID,
                finding.region,
                finding.resource_id,
                unroll_dict(unroll_tags(finding.resource_tags)),
            ):
```

File: prowler/lib/mutelist/mutelist.py [hunk 2]
```diff
@@ -237,6 +238,37 @@ def is_muted_in_check(
...
+                finding.raw["status"] = finding.status
+                finding.status = "MUTED"
+                finding.muted = True
+            return finding
+        except Exception as error:
+            logger.error(
+                f"{error.__class__.__name__} -- {error}[{error.__traceback__.tb_lineno}]"
+            )
+            return False
```
Comment (jfagoagas) [inline prowler/lib/mutelist/mutelist.py]: Why are you returning a `False` instead of the finding?

Comment (jfagoagas) [reply prowler/lib/mutelist/mutelist.py]: Add a test for this case please.

File: tests/providers/aws/lib/mutelist/aws_mutelist_test.py [hunk 3]
```diff
@@ -1844,3 +1844,37 @@ def test_is_muted_in_resource_starting_by_star(self):
...
+                            "Resources": ["prowler", "^test", "prowler-pro"],
+                        }
+                    }
+                }
+            }
+        }
+        mutelist = AWSMutelist(mutelist_content=mutelist_content)
+
+        # Finding
+        finding_1 = MagicMock
```
Comment (jfagoagas) [inline tests/providers/aws/lib/mutelist/aws_mutelist_test.py]: Why are you not using a `Finding`? We have a cool generator for this.

File: tests/providers/aws/lib/mutelist/aws_mutelist_test.py [hunk 4]
```diff
@@ -1844,3 +1844,37 @@ def test_is_muted_in_resource_starting_by_star(self):
...
+        finding_1.region = AWS_REGION_US_EAST_1
+        finding_1.account_uid = AWS_ACCOUNT_NUMBER
+        finding_1.resource_id = "prowler"
+        finding_1.resource_tags = []
+        finding_1.muted = False
+        finding_1.raw = {}
+
+        muted_finding = mutelist.mute_finding(finding_1)
+
+        assert muted_finding.status == "MUTED"
```
Comment (jfagoagas) [inline tests/providers/aws/lib/mutelist/aws_mutelist_test.py:1877]: Please, use the `Status` enum here.

File: prowler/lib/mutelist/mutelist.py [hunk 5]
```diff
@@ -237,6 +238,37 @@ def is_muted_in_check(
...
+                finding.account_uid,
+                finding.metadata.CheckID,
+                finding.region,
+                finding.resource_id,
+                unroll_dict(unroll_tags(finding.resource_tags)),
+            )
+
+            if is_finding_muted:
+                finding.raw["status"] = finding.status
+                finding.status = "MUTED"
```
Comment (jfagoagas) [inline prowler/lib/mutelist/mutelist.py]: Please, use the `Status` enum here, add it if not present.

File: tests/providers/azure/lib/mutelist/azure_mutelist_test.py [hunk 6]
```diff
@@ -66,3 +66,37 @@ def test_is_finding_muted(self):
...
+                            "Resources": ["test_resource"],
+                        }
+                    }
+                }
+            }
+        }
+
+        mutelist = AzureMutelist(mutelist_content=mutelist_content)
+
+        finding = MagicMock
```
Comment (jfagoagas) [inline tests/providers/azure/lib/mutelist/azure_mutelist_test.py]: Why are you not using a `Finding`? We have a cool generator for this.

Comment (jfagoagas) [reply tests/providers/azure/lib/mutelist/azure_mutelist_test.py]: The same for the rest of the tests.

File: tests/providers/aws/lib/mutelist/aws_mutelist_test.py [hunk 7]
```diff
@@ -1844,3 +1844,37 @@ def test_is_muted_in_resource_starting_by_star(self):
...
+        assert muted_finding.muted
+        assert muted_finding.raw["status"] == "FAIL"
```
Comment (jfagoagas) [inline tests/providers/aws/lib/mutelist/aws_mutelist_test.py:1879]: The same for the status enum.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: tests/providers/aws/lib/mutelist/aws_mutelist_test.py (hunk 7 above)
Comment (pedrooot) [reply tests/providers/aws/lib/mutelist/aws_mutelist_test.py:1879]: Done

File: tests/providers/azure/lib/mutelist/azure_mutelist_test.py (hunk 6 above)
Comment (pedrooot) [reply tests/providers/azure/lib/mutelist/azure_mutelist_test.py]: Done

File: prowler/lib/mutelist/mutelist.py (hunk 5 above)
Comment (pedrooot) [reply prowler/lib/mutelist/mutelist.py]: Done

File: tests/providers/aws/lib/mutelist/aws_mutelist_test.py (hunk 4 above)
Comment (pedrooot) [reply tests/providers/aws/lib/mutelist/aws_mutelist_test.py:1877]: Done

File: tests/providers/aws/lib/mutelist/aws_mutelist_test.py (hunk 3 above)
Comment (pedrooot) [reply tests/providers/aws/lib/mutelist/aws_mutelist_test.py]: Done

File: prowler/lib/mutelist/mutelist.py (hunk 1 above)
Comment (pedrooot) [reply prowler/lib/mutelist/mutelist.py]: Done

**Review by jfagoagas — APPROVED** [review]
Awesome, well done!

**Pull request merged**